src/main/java/com/exasol/releasedroid/adapter/repository/GenericRepository.java

```diff
@@ -0,0 +1,63 @@
+package com.exasol.releasedroid.adapter.repository;
+
+import java.util.List;
+import java.util.Map;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.releasedroid.adapter.communityportal.CommunityPlatformValidator;
+import com.exasol.releasedroid.adapter.github.GitHubGateway;
+import com.exasol.releasedroid.adapter.github.GitHubPlatformValidator;
+import com.exasol.releasedroid.adapter.jira.JiraPlatformValidator;
+import com.exasol.releasedroid.usecases.exception.RepositoryException;
+import com.exasol.releasedroid.usecases.repository.BaseRepository;
+import com.exasol.releasedroid.usecases.repository.RepositoryGate;
+import com.exasol.releasedroid.usecases.request.PlatformName;
+import com.exasol.releasedroid.usecases.validate.ReleasePlatformValidator;
+import com.exasol.releasedroid.usecases.validate.RepositoryValidator;
+
+/**
+ * Represents a repository with undefined programming language. Allows providing minimum release functionality for
+ * unsupported programming languages.
+ */
+public class GenericRepository extends BaseRepository {
+    private final List<RepositoryValidator> repositoryValidators = List.of(new CommonRepositoryValidator(this));
+    private final Map<PlatformName, ReleasePlatformValidator> platformValidators;
+
+    /**
+     * Create a new instance of {@link GenericRepository}.
+     *
+     * @param repositoryGate repository gate
+     * @param githubGateway  github gateway
+     */
+    public GenericRepository(final RepositoryGate repositoryGate, final GitHubGateway githubGateway) {
+        super(repositoryGate);
+        this.platformValidators = Map.of( //
+                PlatformName.GITHUB, new GitHubPlatformValidator(this, githubGateway), //
+                PlatformName.COMMUNITY, new CommunityPlatformValidator(this), //
+                PlatformName.JIRA, new JiraPlatformValidator(this) //
+        );
+    }
+
+    @Override
+    public String getVersion() {
```

Comment:
That will fail if the changelog is in the wrong order...
Also, I quite often forget to add my version to change log but only add the changes_x.x.x.md file.
I write a universal comparator for versions here: https://github.com/exasol/error-catalog/blob/main/src/main/java/com/exasol/errorcodecatalog/renderer/ProjectVersion.java
It can sort arbitrary version numbers (also if they include characters).
I'm wondering if we should extract it, since I also need it in project keeper (to validate and generate the changelog file).
On the other hand if all projects would use project-keeper, this solution is fine.
So I'am not sure. You could do a more fancy validation here. But maybe it's also ok like that.

Reply:
But if the changes file contains something like `][` this method will crash.
Maybe you can catch this by checking `from < to` or by using a regex

Reply:
I would keep it simple for now and adjust later if we encounter some problems.  checking from < to sounds good though

Reply:
If the versions are in the wrong order, the validation will fail